Automattic/jetpack
# Social: Use Auto Conversion logic on the UI

This change updates the logic to disable certain connections if the selected media is invalid. Now if auto conversion is enabled, the connections wont get disabled if the conversion is possible

## Proposed changes:

* Added the new logic for handling the conversion
* Updated the notices accordingly

### Other information:


## Jetpack product discussion


https://github.com/orgs/Automattic/projects/733/views/2?pane=issue&itemId=36022365
## Does this pull request change what data or activity we track or use?


No.
## Testing instructions:






* Check out this patch while also having the `jetpack-social-image-auto-convert` blog sticker on your site
  * Turn off the auto-conversion in the Social admin page. All media validation should work as before
    * Connections should get disabled if the media is invalid for them
  * Turn on the auto-conversion
    * Selecting a convertible media that is invalid for a connection, should not disable the connection and the notice should appear stating that it will be converted
    * Selecting a non-convertible media (videos, or images that are more than 55Mb-s) should result the same errors as with auto-conversion disabled.
*
